### ISSUE TITLE
Questions from a newbie: How to setup project

### ISSUE THREAD

**Issue by villmow** [mozilla/bugbug#812]
Hi guys, first of all nice project. I'm a researcher in nlp in the context of software engineering so this projects fits quite nicely. I'm trying to investigate if I can contribute something. 

I'm currently having some issues setting up the project to be able to work on it and understanding how the components work together (I'm new to the mozilla project structure):
 1. Which mercurial version is currently supported?
 2. Should I run the docker images?
 3. How does the data handling work? 
  3.1 Lets say I want to train a new defect classifier, what data will be used? 
  3.2 Is there already a fixed train, validation and test split for comparison? Will that be created during training? Or do you always test on the newest commits? 
  3.3 Where to find the data downloaded when running  `python run.py --train --goal defect`
  3.4 How to get the 100,000 bug reports (as mentioned in the article) from the last 2 years to train on?
  3.5 What about the `repository.py` script? What data will that script download? Can I use that data to train a classifier as well? 

Thanks in advance!

**Comment by ayush-1506** [comment]
Hey. I'll try to answer as many as I can.

1 : Mercurial 4.8.2

3.1 : There are two main data files: data/bugs.json and data/commits.json, that store the bug and commit data. These are the ones used as input to the models.

3.2 : Right now we use train_test_split for randomly getting the train, test data. (See [here](https://github.com/mozilla/bugbug/blob/master/bugbug/model.py#L301))

3.3 : Once you run `python run.py --train --goal=defect`, the data files (as mentioned above) are downloaded in the `data/` folder.

3.4 :
```
from bugbug import bugzilla
bugs = bugzilla.get_bugs()
```
`bugs` here is a generator object that gives you the required bug data. (It reads data from  data/bugs.json).

3.5 : The repository script is used to generate the commits.json data file, by mining data from the mercurial log of the mozilla-central repository. However, it's better to download the commits.json file rather than generate it, since it can take a while for the repository script to run (~ 8-9 hours).

**Comment by villmow** [comment]
Hi, thanks for your answers!

So I understand that the commits.json file is generated by the repository script and you keep an updated version to download. Then I don't really need mercurial right?

> 3.2 : Right now we use train_test_split for randomly getting the train, test data. (See here)
Just putting it together for defects: Under `bugbug/labels/bug_nobug.csv` you have labels for some bugs. For training you're using only the bugs that have labels in that file or does any bug returned by your snippet above have labels?

**Comment by marco-c** [comment]
> Hi guys, first of all nice project. I'm a researcher in nlp in the context of software engineering so this projects fits quite nicely. I'm trying to investigate if I can contribute something.

Hello, and welcome! Are you interested in something in particular?

@ayush-1506 already answered to everything, except:
> Should I run the docker images?

You can if you want (and it can definitely help especially if you want to work on the repository mining script), but it's not required.

Regarding Mercurial, I think we support anything starting from 4.8, but I haven't tested anything newer than that (and on CI and in production we are using 4.8).

**Comment by ayush-1506** [comment]
You don't need mercurial unless you're working on functionalities in the repository script.

The labels in `bugbug/labels/bug_nobug.csv` are ones that have been manually collected. However, the data for the defect model comes from this file, along with identifying some keywords that bugzilla provides (so we iterate through our bugs DB to collect more data points too).

More on this : https://github.com/mozilla/bugbug/blob/master/bugbug/models/defect.py#L87

**Comment by marco-c** [comment]
I've pushed a few commits to clarify some of your questions in the README. I'm closing this but feel free to keep asking questions if you need.
